File: Java/Embed for your customers/AppOwnsData/src/main/java/com/embedsample/appownsdata/models/ReportConfig.java
// ----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
// ----------------------------------------------------------------------------

package com.embedsample.appownsdata.models;

import org.json.JSONException;
import org.json.JSONObject;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.embedsample.appownsdata.controllers.EmbedController;

/**
 * Properties for embedding the report 
 */
public class ReportConfig {
	static final Logger logger = LoggerFactory.getLogger(EmbedController.class);
	
	public String reportId = "";

	public String embedUrl = "";
	
	public String reportName = "";
	
	public Boolean isEffectiveIdentityRolesRequired = false;

	public Boolean isEffectiveIdentityRequired = false;

	public Boolean enableRLS = false;

	public String username;

	public String roles;
	
	public JSONObject getJSONObject() {
		JSONObject jsonObj = new JSONObject();
		try {
			jsonObj.put("reportId", reportId);
			jsonObj.put("embedUrl", embedUrl);
			jsonObj.put("reportName", reportName);
		} catch (JSONException e) {
			logger.error("DefaultListItem.toString JSONException: " + e.getMessage());
		}
		return jsonObj;
	}
}

File: Python/Embed for your customers/AppOwnsData/models/reportconfig.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT license

class ReportConfig:

    # Camel casing is used for the member variables as they are going to be serialized and camel case is standard for JSON keys

    reportId = None
    reportName = None
    embedUrl = None
    datasetId = None

    def __init__(self, report_id, report_name, embed_url, dataset_id = None):
        self.reportId = report_id
        self.reportName = report_name
        self.embedUrl = embed_url
        self.datasetId = dataset_id